app/lib/utils/apiResponse.ts

```diff
@@ -0,0 +1,17 @@
+import { NextResponse } from 'next/server';
+
+interface ErrorResponse {
+  success: false;
+  errors: string[] | Record<string, unknown>;
+}
+
+export const successResponse = <T>(data: T, status: number = 200) => {
```

Comment:
If you don't use the `data` parameter inside this generic function and just pass it into the `.json` function, you can omit the generic at all. 
```suggestion
export const successResponse = (data: unknown, status: number = 200) => {
```